### PR TITLE
feat: Allow buildkit timeout to be configurable

### PR DESCRIPTION
**Pull request by terrytangyuan**
This was needed for me to get everything to run. I'd imagine other users might benefit from this as well.

### CODE REVIEW

File: pkg/util/envutil/env_test.go [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLookupEnvDurationOr(t *testing.T) {
+	defer func() { _ = os.Unsetenv("FOO") }()
+	assert.Equal(t, time.Second, LookupEnvDurationOr("", time.Second), "default value")
+	_ = os.Setenv("FOO", "bar")
```
Comment (hezhizhen) [inline pkg/util/envutil/env_test.go:28]: use `t.Setenv()` in tests.

And the `defer` will be no longer needed.

File: pkg/util/envutil/env.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+package envutil
+
+import (
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func LookupEnvDurationOr(key string, o time.Duration) time.Duration {
```
Comment (hezhizhen) [inline pkg/util/envutil/env.go:24]: How about `GetDurationWithDefault`? I think there is no need to emphasize the action `lookup env`.

File: pkg/util/envutil/env.go [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+	v, found := os.LookupEnv(key)
+	if found && v != "" {
+		d, err := time.ParseDuration(v)
+		if err != nil {
+			log.WithField(key, v).WithError(err).Panic("failed to parse")
+		} else {
```
Comment (hezhizhen) [inline pkg/util/envutil/env.go:30]: Handle this case first because it returns directly.

**Comment by terrytangyuan** [comment]
@hezhizhen Great suggestions! Updated.

File: pkg/util/envutil/env.go [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+package envutil
+
+import (
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+)
+
+func GetDurationWithDefault(key string, o time.Duration) time.Duration {
```
Comment (hezhizhen) [inline pkg/util/envutil/env.go:24]: It's better to give some explanation about accepted time syntax in comments.

File: pkg/util/envutil/env_test.go [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestGetDurationWithDefault(t *testing.T) {
+	assert.Equal(t, time.Second, GetDurationWithDefault("", time.Second), "default value")
+	t.Setenv("FOO", "bar")
+	assert.Panics(t, func() { GetDurationWithDefault("FOO", time.Second) }, "bad value")
+	t.Setenv("FOO", "1h")
```
Comment (hezhizhen) [inline pkg/util/envutil/env_test.go:28]: How about using the actual environment variable name?

File: pkg/util/envutil/env.go [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+	v, found := os.LookupEnv(key)
+	if found && v != "" {
+		d, err := time.ParseDuration(v)
+		if err == nil {
+			return d
+		}
+		log.WithField(key, v).WithError(err).Panic("failed to parse")
```
Comment (hezhizhen) [inline pkg/util/envutil/env.go:31]: will `Panic()` panic here?

Comment (terrytangyuan) [reply pkg/util/envutil/env.go:31]: Yes it's intentional since we don't want users to assume that the incorrect config is applied when actually not.

**Comment by gaocegege** [comment]
PTAL @hezhizhen

**Comment by hezhizhen** [comment]
LGTM

**Comment by gaocegege** [comment]
Thanks for your contribution! :tada: :+1:

And Thank @hezhizhen for the review.

**Pull request merged**